addon/components/relationship-select.js

```diff
@@ -0,0 +1,23 @@
+import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
+import { timeout } from "ember-concurrency";
+import { restartableTask, lastValue } from "ember-concurrency-decorators";
+
+import handleModelErrors from "ember-emeis/decorators/handle-model-errors";
+
+export default class RelationshipSelectComponent extends Component {
+  @service notification;
+  @service intl;
+  @service store;
+
+  @lastValue("fetchModels") models;
+
+  @restartableTask
+  @handleModelErrors
+  *fetchModels(search) {
+    yield timeout(200);
```

Comment:
Maybe only if you actually have a search? Otherwise we already debounce the initial request. Also, `200ms` is not enough debouncing for a normal typing speed

Reply:
Done :+1: